src/wallet/api/wallet.cpp

```diff
@@ -3154,7 +3154,47 @@ void WalletImpl::processPoolState(const std::vector<std::tuple<cryptonote::trans
     }
 }
 //-------------------------------------------------------------------------------------------------------------------
-std::string convertMultisigTxToStr(const PendingTransaction &multisig_ptx)
+void WalletImpl::getEnoteDetails(std::vector<EnoteDetails> enote_details) const
+{
+    wallet2::transfer_container tc;
+    m_wallet->get_transfers(tc);
+    enote_details.reserve(tc.size());
+
+    for (auto &td : tc)
+    {
+        EnoteDetails ed{};
+
+        cryptonote::txout_target_v txout_v = td.m_tx.vout[tx.m_internal_output_index];
+        crypto::public_key pub_key_pod;
+        if (txout_v.type() == typeid(cryptonote::txout_to_key))
+            ed.m_onetime_address = epee::string_tools::pod_to_hex(boost::get<cryptonote::txout_to_key>(txout_v).key);
+        else if (txout_v.type() == typeid(cryptonote::txout_to_tagged_key))
+        {
+            ed.m_onetime_address = epee::string_tools::pod_to_hex(boost::get<cryptonote::txout_to_tagged_key>(txout_v).key);
+            ed.m_view_tag = epee::string_tools::pod_to_hex(boost::get<cryptonote::txout_to_tagged_key>(txout_v).view_tag);
+        }
+
+        ed.m_tx_id = td.m_txid;
+        ed.m_internal_output_index = td.m_internal_output_index;
+        ed.m_global_output_index = td.m_global_output_index;
+        ed.m_spent = td.m_spent;
+        ed.m_frozen = td.m_frozen;
+        ed.m_spent_height = td.m_spent_height;
+        ed.m_key_image = td.m_key_image;
+        ed.m_mask = td.m_mask;
+        ed.m_amount = td.m_mask;
+        ed.m_rct = td.m_rct;
```

Comment:
Shouldn't we use here the same enum we have at the transaction, soon with 3 values?